crates/ruff/src/importer/insertion.rs

```diff
@@ -288,6 +284,18 @@ fn match_docstring_end(body: &[Stmt]) -> Option<TextSize> {
     Some(stmt.end())
 }
 
+/// If a line starts with a semicolon, return its offset.
+fn match_leading_semicolon(s: &str) -> Option<TextSize> {
+    for (offset, c) in s.char_indices() {
+        match c {
+            ' ' | '\t' => continue,
+            ';' => return Some(TextSize::try_from(offset).unwrap()),
```

Comment:
Is it okay that this returns a relative offset from the beginning of the string?

Reply:
Yup!